src/ray/rpc/grpc_server.h

```diff
@@ -118,6 +148,8 @@ class GrpcServer {
   /// Listen to localhost (127.0.0.1) only if it's true, otherwise listen to all network
   /// interfaces (0.0.0.0)
   const bool listen_to_localhost_only_;
+  /// Token representing ID of this cluster.
+  std::atomic<ClusterID> cluster_id_;
```

Comment:
Could you add some comments why it needs to be atomic?

Reply:
Hmm, also interesting, that ClusterID can be used in std::atomic. I thought only primitives can be atomic.

Reply:
It needs to satisfy `is_trivially_copiable` -- seems clang on OSX still compiles with this, interestingly. Changed to mutex.

Reply:
Also, you are right that a mutex is not necessary here, since SetClusterID will only be called in a reply callback, which is currently run on the same thread as the main GRPC service. However, since acquiring a free mutex is low-cost anyway, it seems safer to add this now in anticipation of a change to the threading model, particularly since completion queue callbacks could easily be run in a separate IO threadpool.